client/homescreen/activity-panel/orders/utils.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { SETTINGS_STORE_NAME, ITEMS_STORE_NAME } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { DEFAULT_ACTIONABLE_STATUSES } from '../../../analytics/settings/config';
+
+export function getUnreadOrders( select ) {
+	const { getItems, getItemsTotalCount, getItemsError, isResolving } = select(
+		ITEMS_STORE_NAME
+	);
+	const { getSetting: getMutableSetting } = select( SETTINGS_STORE_NAME );
+	const {
+		woocommerce_actionable_order_statuses: orderStatuses = DEFAULT_ACTIONABLE_STATUSES,
+	} = getMutableSetting( 'wc_admin', 'wcAdminSettings', {} );
+
+	if ( ! orderStatuses.length ) {
+		return false;
+	}
+
+	const ordersQuery = {
+		page: 1,
+		per_page: 1, // Core endpoint requires per_page > 0.
+		status: orderStatuses,
+		_fields: [ 'id' ],
+	};
+
+	getItems( 'orders', ordersQuery );
+
+	// Disable eslint rule requiring `latestNote` to be defined below because the next two statements
+	// depend on `getItemsTotalCount` to have been called.
+	// eslint-disable-next-line @wordpress/no-unused-vars-before-return
```

Comment:
What do you think about refactoring this a little bit and moving it out of this area? It really should be a selector inside `@woocommerce/data` (in the items store). The only thing that will need to change to accomodate that is to pass in the `orderStatuses` to the selector, that way it's not dependent on the settings store directly to work.

Ideally we would also cover some of this with tests although that will probably require significant mocking right now, so optional imo, but quite important I think to start moving hidden bits of code like this to packages where they belong because it should be reusable.

Reply:
I like the idea, if it's ok with you I will do this refactor in a follow-up PR.

Reply:
Hi @samueljseay,
In order to address this fix, I created the commit 558cdd9, could you give it a look?